pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -2082,6 +2119,67 @@ func TestGetTaskRunTimeout(t *testing.T) {
 	}
 }
 
+func TestGetFinallyTaskRunTimeout(t *testing.T) {
+	prName := "pipelinerun-finallyTimeouts"
+	ns := "foo"
+	p := "pipeline"
+
+	tcs := []struct {
+		name     string
+		pr       *v1beta1.PipelineRun
+		rprt     *resources.ResolvedPipelineRunTask
+		expected *metav1.Duration
+	}{{
+		name: "nil timeout duration",
+		pr: tb.PipelineRun(prName, tb.PipelineRunNamespace(ns),
+			tb.PipelineRunSpec(p, tb.PipelineRunNilTimeout),
+			tb.PipelineRunStatus(tb.PipelineRunStartTime(time.Now())),
+		),
+		rprt: &resources.ResolvedPipelineRunTask{
+			PipelineTask: &v1beta1.PipelineTask{},
+		},
+		expected: &metav1.Duration{Duration: 60 * time.Minute},
+	}, {
+		name: "timeout specified in pr",
+		pr: tb.PipelineRun(prName, tb.PipelineRunNamespace(ns),
+			tb.PipelineRunSpec(p, tb.PipelineRunTimeout(20*time.Minute)),
+			tb.PipelineRunStatus(tb.PipelineRunStartTime(time.Now())),
+		),
+		rprt: &resources.ResolvedPipelineRunTask{
+			PipelineTask: &v1beta1.PipelineTask{},
+		},
+		expected: &metav1.Duration{Duration: 20 * time.Minute},
+	}, {
+		name: "taskrun being created after timeout expired",
+		pr: tb.PipelineRun(prName, tb.PipelineRunNamespace(ns),
+			tb.PipelineRunSpec(p, tb.PipelineRunTimeout(1*time.Minute)),
+			tb.PipelineRunStatus(tb.PipelineRunStartTime(time.Now().Add(-2*time.Minute)))),
+		rprt: &resources.ResolvedPipelineRunTask{
+			PipelineTask: &v1beta1.PipelineTask{},
+		},
+		expected: &metav1.Duration{Duration: 1 * time.Second},
+	}, {
+		name: "40m timeout duration, 20m taskstimeout duration",
+		pr: tb.PipelineRun(prName, tb.PipelineRunNamespace(ns),
+			tb.PipelineRunSpec(p, tb.PipelineRunTimeout(40*time.Minute), tb.PipelineRunTasksTimeout(20*time.Minute)),
+			tb.PipelineRunStatus(tb.PipelineRunStartTime(time.Now())),
+		),
+		rprt: &resources.ResolvedPipelineRunTask{
+			PipelineTask: &v1beta1.PipelineTask{},
+		},
+		expected: &metav1.Duration{Duration: 40 * time.Minute},
+	},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			if d := cmp.Diff(getFinallyTaskRunTimeout(context.TODO(), tc.pr, tc.rprt), tc.expected); d != "" {
```

Comment:
The order of params to `Diff` need to be the other way around so that `PrintWantGot` displays the correct thing in its message; the order goes `(expected, received)`.

Reply:
I'm not sure to get it.

For this test function I get this kind of error
```go
go test -run TestGetFinallyTaskRunTimeout
--- FAIL: TestGetFinallyTaskRunTimeout (0.00s)
    --- FAIL: TestGetFinallyTaskRunTimeout/timeout_specified_in_pr (0.00s)
        pipelinerun_test.go:2177: Unexpected finally task run timeout. Diff (-want, +got):   &v1.Duration{
            -   Duration: s"20m0s",
            +   Duration: s"2m0s",
              }
FAIL
exit status 1
```

The display seems good.

Reply:
Looking at an example test case:

```go
{
	name: "40m timeout duration, 20m taskstimeout duration",
	pr: tb.PipelineRun(prName, tb.PipelineRunNamespace(ns),
		tb.PipelineRunSpec(p, tb.PipelineRunTimeout(40*time.Minute), tb.PipelineRunTasksTimeout(20*time.Minute)),
		tb.PipelineRunStatus(tb.PipelineRunStartTime(time.Now())),
	),
	rprt: &resources.ResolvedPipelineRunTask{
		PipelineTask: &v1beta1.PipelineTask{},
	},
	expected: &metav1.Duration{Duration: 20 * time.Minute},
},
```

I changed the last test case to expect 20 minutes instead of 40, which causes the following error:

```yaml
pipelinerun_test.go:2177: Unexpected finally task run timeout. Diff (-want, +got):  &v1.Duration{
            -   Duration: s"40m0s",
            +   Duration: s"20m0s",
}
```

Here ^ the `(-want, +got)` says that the test "wanted" `- Duration: s"40m0s"` and that it "got" `+ Duration: s"20m0s"`.  In fact we "wanted" 20m0s and "got" 40m0s, so the meaning is flipped.  Does that make sense?

Reply:
Yeah make sense 😊  Thanks for reviewing this.